### DISCUSSION TITLE
How to plot individual components of a signal using TimeSeries.plot

### DISCUSSION THREAD

**Discussion by gaozelle** [kineticstoolkit/kineticstoolkit#138] [1. Users Q&A] [answered]
Hello
I'm using kineticstoolkit to extract some biomechanical data from C3D. But I found that using the plot() function it is not possible to select separate coordinates or angle data.

like this:
  
```python
markers.plot(['LHipAngles', 'LKneeAngles', 'RGroundReactionForce'])
```
![image](https://user-images.githubusercontent.com/31237218/206842035-4df1914c-b7c3-40fe-b9bd-661c931652a6.png)
  
The plot() function plots all the data in the plot.
  
Please tell me if there is a way to extract the individual values of the z-axis of ['LHipAngles', 'LKneeAngles', 'RGroundReactionForce']  , 
like : (['LHipAngles[2]', 'LKneeAngles[2]', 'RGroundReactionForce[2]]) and how to draw them on the image.

Thanks for your help
Thanks for your contribution to kineticstoolkit

**Comment by felixchenier** [comment, marked as answer]
Hello @gaozelle 

The `TimeSeries.plot` function is a somewhat basic helper function that uses Matplotlib to quickly see selected signals with events. There is no option to extract a given component of a signal. If you want more control on what you want to visualize, you can use `matplotlib.pyplot.plot` directly. Assuming (maybe wrongly) some labels for your data:

```
plt.figure()
plt.plot(markers.time, markers.data['LHipAngles'][:, 2], label="Hip flexion")
plt.plot(markers.time, markers.data['LKneeAngles'][:, 2], label="Knee extension")
plt.plot(markers.time, markers.data['RGroundReactionForce'][:, 2], label="Ground reaction force")
plt.legend()
...
```

Or maybe better yet, you could generate a new TimeSeries that contains only the data that you need:

```
ts = markers.copy(copy_data=False)
ts.data['HipFlexion'] = markers.data['LHipAngles'][:, 2]
ts.data['KneeExtension'] = markers.data['LKneeAngles'][:, 2]
ts.data['GroundReactionForce'] = markers.data['RGroundReactionForce'][:, 2]

ts.plot()
```

The second method has the benefit of also printing the events.

Hope it helps you!
